packages/core/src/state-management/repository.ts

```diff
@@ -531,8 +531,8 @@ export class Repository {
       this.logger.verbose(`Learned of new tip ${cid} for stream ${state$.id}`)
       const next = await this.streamUpdater.applyTipFromNetwork(state$.state, cid)
       if (next) {
-        await this._updateStateIfPinned(state$)
         state$.next(next)
+        await this._updateStateIfPinned(state$)
```

Comment:
huh how does this fix the problem we saw, I'm not deep in enough to understand exactly - is it that we didn't advance the state before pinning it?

Reply:
we write the old, pre-anchor-commit state to the state store.

This code is applying a new commit, but writing the old state that doesn't contain the new commit to the state store.  This fixes the code to apply the commit to the state and THEN update the state store with the new state